src/View/SSTemplateParser.peg

```diff
@@ -1035,23 +1026,23 @@ class SSTemplateParser extends Parser implements TemplateParser
                 'arguments only.', $this);
         }
 
-        //loop without arguments loops on the current scope
+        // loop without arguments loops on the current scope
         if ($res['ArgumentCount'] == 0) {
-            $on = '$scope->locally()->obj(\'Me\', [], true)';
+            $on = '$scope->locally()->self()';
         } else {    //loop in the normal way
             $arg = $res['Arguments'][0];
             if ($arg['ArgumentMode'] == 'string') {
                 throw new SSTemplateParseException('Control block cant take string as argument.', $this);
             }
             $on = str_replace(
                 '$$FINAL',
-                'obj',
+                'scopeToIntermediateValue',
                 ($arg['ArgumentMode'] == 'default') ? $arg['lookup_php'] : $arg['php']
             );
         }
 
         return
-            $on . '; $scope->pushScope(); while (($key = $scope->next()) !== false) {' . PHP_EOL .
+            $on . '; $scope->pushScope(); while ($scope->next() !== false) {' . PHP_EOL .
```

Comment:
`$key` was never used. I wasted a bunch of time trying to find where it's used. I've removed that variable to avoid wasting that time again next time I look at this.